lib/datacite/register_doi.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Datacite
+  # The JSON request to create a specific DOI
+  class RegisterDoi
```

Comment:
I would expect a class named "RegisterDoi" to do the actual registration.  Perhaps `DoiRequestJSON` ?  or even `DOIRequestJSON`  or just plain `DOIRequest`?

Reply:
I think that "register" has to be in there somewhere, so as not to confuse it with retrieving a DOI.

Reply:
I don't like `JSON` in the name because this object is not the JSON, but has a method that produces JSON.

Reply:
So... `RegisterDoiRequestBodyFactory`?

Reply:
I think a request for a DOI is different from retrieving a DOI.   But having a method in one class that is named "registerDOI" ... that actually does the registration, and a class named "RegisterDOI" that does NOT do the registration ... smells.

Reply:
`RegisterDoiRequestBody` or with the factory is clearer, yes.